docs/reference-guides/block-api/block-metadata.md

```diff
@@ -632,7 +632,7 @@ return array(
 Starting in the WordPress 5.8 release, it is possible to instruct WordPress to enqueue scripts and styles for a block type only when rendered on the frontend. It applies to the following asset fields in the `block.json` file:
 
 -   `script`
--   `viewScript` (when the block defines `render_callback` during registration in PHP, then the block author is responsible for enqueuing the script)
+-   `viewScript` (when the block defines `render_callback` during registration in PHP or a `render` field in its `block.json`, then the script is registered but the block author is responsible for enqueuing it)
```

Comment:
Good catch 👍🏻 

Aside: Sharing for inspiration. It would be great to remove that limitation and offer an alternative way to enqueue those view scripts on demand. Relate Track ticket: https://core.trac.wordpress.org/ticket/56470.

Reply:
Yes, something like that would be neat 🙂